test/Serialization/ToricGeometry.jl

```diff
@@ -2,9 +2,11 @@
   mktempdir() do path
     @testset "NormalToricVariety" begin
       pp = projective_space(NormalToricVariety, 2)
-      test_save_load_roundtrip(path, pp) do loaded
+      R = cox_ring(pp)
+      test_save_load_roundtrip(path, pp; with_attrs=true) do loaded
         @test rays(pp) == rays(loaded)
         @test ray_indices(maximal_cones(pp)) == ray_indices(maximal_cones(loaded))
+        @test R == get_attribute(loaded, :cox_ring)
```

Comment:
since `NormalToricVariety` serializes with `uses_id`, this test does not test anything here, as already `pp === loaded` (cf. our discussion in Durham)

Reply:
yes, I also realized this must be the case since the fix with the set_attribute -> set_attribute!

Reply:
i added a fix to actually test this now